src/Raven.Server/Documents/Subscriptions/LiveSubscriptionPerformanceCollector.cs

```diff
@@ -0,0 +1,305 @@
+                        connectionAggregators.AddRange(connectionState.RecentRejectedConnections.Select(x => x.GetPerformanceStats()));
+
+                        // add connection stats to results 
+                        var subscriptionItem = results.Find(x => x.TaskId == kvp.Value.Handler.Connection?.SubscriptionId);
+                        if (subscriptionItem != null)
+                        {
+                            var connectionPerformance = connectionAggregators.Select(x => x.ToConnectionPerformanceLiveStatsWithDetails());
+                            subscriptionItem.ConnectionPerformance = connectionPerformance.ToArray();
+                        }
+                    }
+                }
+
+                foreach (var kvp in _perSubscriptionBatchStats)
+                {
+                    var connectionState = Database.SubscriptionStorage.GetSubscriptionConnection(context, kvp.Value.Handler.SubscriptionName);
+                    if (connectionState != null)
+                    {
+                        var currentConnection = connectionState.Connection;
+                        var batchesAggregators = new List<SubscriptionBatchStatsAggregator>();
+
+                        if (currentConnection != null)
+                        {
+                           // add batches history for inProgress connection
+                           batchesAggregators.AddRange(currentConnection.GetBatchesPerformanceStats());
+                        }
+                       
+                        // add batches history for previous connections
+                        foreach (var recentConnection in connectionState.RecentConnections)
+                        {
+                            batchesAggregators.AddRange(recentConnection.GetBatchesPerformanceStats());
+                        }
+                        foreach (var recentRejectedConnection in connectionState.RecentRejectedConnections)
+                        {
+                            batchesAggregators.AddRange(recentRejectedConnection.GetBatchesPerformanceStats());
+                        }
+                       
+                        // add batch stats to results
+                        var subscriptionItem = results.Find(x => x.TaskId == kvp.Value.Handler.Connection?.SubscriptionId);
+                        if (subscriptionItem != null)
+                        {
+                            var batchPerformance = batchesAggregators.Select(x => x.ToBatchPerformanceLiveStatsWithDetails());
+                            subscriptionItem.BatchPerformance = batchesAggregators.Select(x => x.ToBatchPerformanceLiveStatsWithDetails()).ToArray();
+                        }
+                    }
+                }
+            }
+            
+            return results;
+        }
+
+        protected override async Task StartCollectingStats()
+        {
+            Database.SubscriptionStorage.OnEndConnectionEvent += OnEndConnection;
```

Comment:
On (Subject) (Verb/Action) Event vs On (Verb/Action) (Subject) Event 

Can we unify event names?

Reply:
done
```
Database.SubscriptionStorage.OnEndConnectionEvent += OnEndConnection;
Database.SubscriptionStorage.OnEndBatchEvent += OnEndBatch;
Database.SubscriptionStorage.OnAddTaskEvent += OnAddSubscriptionTask;
Database.SubscriptionStorage.OnRemoveTaskEvent += OnRemoveSubscriptionTask;
```